cuML/src/glm/ols.h

```diff
@@ -94,23 +85,14 @@ void olsFit(math_t *input, int n_rows, int n_cols, math_t *labels, math_t *coef,
 	}
 
 	if (fit_intercept) {
-		postProcessData(input, n_rows, n_cols, labels, coef, intercept, mu_input,
-				mu_labels, norm2_input, fit_intercept, normalize, cublas_handle,
-				cusolver_handle);
-
-		if (normalize) {
-			if (norm2_input != NULL)
-				cudaFree(norm2_input);
-		}
-
-		if (mu_input != NULL)
-			cudaFree(mu_input);
-		if (mu_labels != NULL)
-			cudaFree(mu_labels);
+            postProcessData(input, n_rows, n_cols, labels, coef, intercept, mu_input.data(),
+                            mu_labels.data(), norm2_input.data(), fit_intercept, normalize,
+                            cublas_handle, cusolver_handle);
 	} else {
 		*intercept = math_t(0);
 	}
-
+    CUDA_CHECK(cudaStreamSynchronize(s_));
+    CUDA_CHECK(cudaStreamDestroy(s_));
```

Comment:
`cudaStreamDestroy` should not be necessary if `cumlHandle` is passed in.

Reply:
Done.